CryptKeeperZK/crypt-keeper-extension
# feat: support disconnect wallet

## Explanation

This PR adds support for disconnecting user wallet and lock without connecting wallet.
If user clicks "Disconnect wallet" then eager connect stops working until user connects to wallet. 

Details are below:
- [x] User can disconnect wallet and connect it again without automatic reconnection
- [x] User can lock extension if wallet is not connected
- [x] Moved connect/disconnect logic to `useWallet` hook
- [x] Add storage to persist some general purpose values

## More Information

Closes #2

## Screenshots/Screencaps

### Before

![image](https://user-images.githubusercontent.com/14254374/223547364-e065cbb3-eecc-47e0-8547-2f96d062e499.png)

![image](https://user-images.githubusercontent.com/14254374/223547483-f2df8455-be45-4ffa-b895-42e8f3ccf320.png)

### After

![image](https://user-images.githubusercontent.com/14254374/223545413-4a72ef66-9443-4880-acfd-dacafa851061.png)

![image](https://user-images.githubusercontent.com/14254374/223545461-8994cd6e-dbe0-4291-8d23-571085e21338.png)

## Manual Testing Steps

1. Try to use different combinations for "Connect", "Disconnect" and "Lock" actions
2. After "Disconnect" check that there is no automatic connect

## Pre-Merge Checklist


### + If there are functional changes:


> PR template source from [github.com/MetaMask](https://github.com/MetaMask)
